Re-org and revamp k3s etcd-snapshot docs (#447)
k3s-io/docs: docs/cli/etcd-snapshot.md

# k3s etcd-snapshot

This page describes how to use the `k3s etcd-snapshot` CLI tool to manage etcd snapshots and how to restore from an etcd snapshot.

> K3s snapshots are stored in the node file system and optionally be replicated in an S3 compatible object store for disaster recovery scenarios. Moreover, K3s snapshots can be scheduled or taken manually on-demand.  The `k3s etcd-snapshot` CLI tool supports a set of subcommands for working with both your etcd scheduled or on-demand snapshots, as well as local or S3 stored snapshots.

Reviewer: Simplified the language here a bit. We should also be consistent in calling these etcd snapshots, not k3s snapshots.

```suggestion
K3s etcd snapshots are stored on the node file system, and may optionally be uploaded to an S3 compatible object store for disaster recovery scenarios. Snapshots can be scheduled or taken manually on-demand. The `k3s etcd-snapshot` CLI tool offers a set of subcommands that can be used to create, delete, and manage snapshots.
```

Author: looks good, thanks Brad

| Subcommand | Description |
| ----------- | --------------- |
| delete      |  Delete given snapshot(s) |
| ls, list, l |  List snapshots |
| prune       |  Remove snapshots that exceed the configured retention count |
| save        |  Trigger an on-demand etcd snapshot |

For additional information on the etcd snapshot subcommands, run `k3s etcd-snapshot --help`.

> ## Creating snapshots

Reviewer: The convention is for section titles to be title-case

```suggestion
## Creating Snapshots
```

<Tabs groupId="snapshots">
<TabItem value="Scheduled">

Scheduled snapshots are enabled by default, at 00:00 and 12:00 system time, with 5 snapshots retained. Scheduled snapshots have a name that starts with `etcd-snapshot`, followed by the node name and timestamp.

> These are the configuration options that we can set for scheduled snapshots:

Reviewer: this is a little informal, how about:
```suggestion
The following options control the operation of scheduled snapshots:
```

Author: good

| `--etcd-snapshot-compress` | Compress etcd snapshots |
| `--etcd-snapshot-dir` | Directory to save db snapshots. (Default location: `${data-dir}/db/snapshots`) |
| `--etcd-snapshot-retention` | Number of snapshots to retain (default: 5) |
| `--etcd-snapshot-schedule-cron` |  Snapshot interval time in cron spec. eg. every 5 hours `0 */5 * * *` (default: `0 */12 * * *`) |

The data-dir value defaults to `/var/lib/rancher/k3s` and can be changed independently by setting the `--data-dir` flag.

Scheduled snapshots are saved to the path set by the server's `--etcd-snapshot-dir` value. If you want them replicated in S3 compatible object stores, refer to [S3 configuration options](https://docs.k3s.io/cli/etcd-snapshot#s3-compatible-object-store-support)

</TabItem>
<TabItem value="On-demand">

Snapshots can be saved manually by running the `k3s etcd-snapshot save` command. There is no retention for these on-demand snapshots and the user needs to remove them manually by using `k3s etcd-snapshot delete` or `k3s etcd-snapshot prune` commands. On-demand snapshots have a name that starts with `on-demand`, followed by the node name and timestamp.

> These are the configuration options that we can set for on-demand snapshots:

Reviewer: ```suggestion
The following options control the operation of on-demand snapshots:
```

  --s3-secret-key=<S3-SECRET-KEY>
```

```bash
k3s etcd-snapshot delete          \
  --s3                            \
  --s3-bucket=<S3-BUCKET-NAME>    \
  --s3-access-key=<S3-ACCESS-KEY> \
  --s3-secret-key=<S3-SECRET-KEY> \
  <SNAPSHOT-NAME>
```

As snapshots are replicated in S3, these commands will create and delete two snapshots: one stored in the file system and one in S3.

> 

Reviewer 2: This line is confusing. Are you saying it creates 2 copies of the snapshot, upload one and deletes the other?

Reviewer: Yeah, this is confusing as written. I would probably change this to an example command with output using the `shell-session` block style, as we do for the `$ kubectl describe etcdsnapshotfile` examples below.

Select the tab below that matches your cluster configuration.

<Tabs queryString="etcdsnap">

> <TabItem value="Single Server" default>

Reviewer 2: Nit: a default tab is not required if its the first tab defined.

Author: I did not know this, thanks
